doc/Data Plane Code Generation/ServiceDrivenEvolution.md

```diff
@@ -0,0 +1,177 @@
+    request.Headers.Add("Content-Type", "application/json");
+    request.Content = content;
+    return message;
+}
+```
+
+Generated code in a V2 client with `image/jpeg` as a new content type in addition to `application/json`:
+
+``` C#
+...
+/// <param name="contentType"> Body Parameter content-type. Allowed values: &quot;application/json&quot; | &quot;image/jpeg&quot;. </param>
+...
+public virtual async Task<Response> PostParametersAsync(RequestContent content, ContentType contentType, RequestContext context = null)
+{
+    Argument.AssertNotNull(content, nameof(content));
+
+    using var scope = ClientDiagnostics.CreateScope("ParamsClient.PostParameters");
+    scope.Start();
+    try
+    {
+        using HttpMessage message = CreatePostParametersRequest(content, contentType, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    catch (Exception e)
+    {
+        scope.Failed(e);
+        throw;
+    }
+}
+
+internal HttpMessage CreatePostParametersRequest(RequestContent content, ContentType contentType, RequestContext context)
+{
+    var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+    var request = message.Request;
+    request.Method = RequestMethod.Post;
+    var uri = new RawRequestUriBuilder();
+    uri.Reset(_endpoint);
+    uri.AppendPath("/serviceDriven/parameters", false);
+    request.Uri = uri;
+    request.Headers.Add("Accept", "application/json");
+    request.Headers.Add("Content-Type", contentType.ToString());
+    request.Content = content;
+    return message;
+}
+```
+
+Since the V2 client added a required parameter, it breaks the code that is written for V1. We can manually add an overload method that has the same signature as V1 to make it backward compatible:
+
+``` C#
+[EditorBrowsable(EditorBrowsableState.Never)]
+public virtual async Task<Response> PostParametersAsync(RequestContent content, RequestContext context = null)
+{
+    Argument.AssertNotNull(content, nameof(content));
+
+    using var scope = ClientDiagnostics.CreateScope("ParamsClient.PostParameters");
+    scope.Start();
+    try
+    {
+        using HttpMessage message = CreatePostParametersRequest(content, ContentType.ApplicationJson, context);
+        return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+    }
+    catch (Exception e)
+    {
+        scope.Failed(e);
+        throw;
+    }
+}
+```
+
+Notice that we leverage the `CreatePostParametersRequest` method in V2 but always pass in the `ContentType.ApplicationJson` as the `contentType` parameter in the manual method to make the behavior align with V1. We can add `[EditorBrowsable(EditorBrowsableState.Never)]` to the manual method to make it invisible in IDE if we don't want to confuse new customers that never used V1 with this overload method that uses a hard-coded content type in addition to the generated method that uses a `contentType` parameter.
```

Comment:
For this part:

> We can add `[EditorBrowsable(EditorBrowsableState.Never)]` to the manual method to make it invisible in IDE if we don't want to confuse new customers that never used V1 with this overload method that uses a hard-coded content type in addition to the generated method that uses a `contentType` parameter.

It would be good to call out that the business rules for the service are what would determine whether or not to add `[EditorBrowsable(EditorBrowsableState.Never)]`.  If the service added the new contentType, for example, to improve performance, and wants to discourage callers from using the old contentType, then they would definitely want to add `[EditorBrowsable(EditorBrowsableState.Never)]` to the old method so new users wouldn't discover it.  If the old contentType remains the default preference, they wouldn't want to add `[EditorBrowsable(EditorBrowsableState.Never)]` to make it easy for users to discover the default.